### DISCUSSION TITLE
FireFox and <form>-Tag

### DISCUSSION THREAD

**Discussion by Joggel72** [mebjas/html5-qrcode#413] [Ideas]
I'm using the scanner inside a form-Tag.
It works fine with Chrome.
But with FireFox it does an submit.

I've check it and found that the buttons have not type="button", so a click will do a submit.
Actual I don't know why it works on Chrome.

I would like to have closer look on getting it to work with Firefox inside a form-Tag.
But I want to participiate on your further development cause it is the best solution I've found.

So if I get it fixed, is it possible that the changes will find a way in the next update and what will be the best way to do this.

Best regards,
Joggel

**Comment by Joggel72** [comment]
If found the issue #409 from user [ssoec4u](https://github.com/ssoec4u) with the same problem

**Comment by Joggel53** [comment]
The fix by expanding the javascript file was successfull.
I've added the attribute type with value "button" to buttons created by the script with createElement("button"):
- camera permisson
- start scanning
- stop scanning


Example:
                n = document.createElement("button");
                n.setAttribute("type", "button"); // Added type button

**Reply by mebjas** [reply]
Thanks for the fix, can you send a PR for this?

**Reply by Joggel72** [reply]
I've done the fix in the generated javascript not in typescript.
I will check the typescript and then do the PR.

**Reply by Joggel72** [reply]
PR is done.
